entity-service-api/src/main/proto/org/hypertrace/entity/constants/v1/entity_attribute.proto

```diff
@@ -27,28 +27,14 @@ enum ApiAttribute {
   API_ATTRIBUTE_ID = 2 [(string_value) = "API_ID"];
   API_ATTRIBUTE_URL_PATTERN = 3 [(string_value) = "API_URL_PATTERN"];
   API_ATTRIBUTE_API_TYPE = 4 [(string_value) = "API_TYPE"];
-  API_ATTRIBUTE_HTTP_METHOD = 5 [(string_value) = "http_method"];
-  API_ATTRIBUTE_PATH_PARAMS_SCHEMA = 6 [(string_value) = "path_params_schema"];
-  API_ATTRIBUTE_QUERY_PARAMS_SCHEMA = 7 [(string_value) = "query_params_schema"];
-  API_ATTRIBUTE_REQUEST_HEADERS = 8 [(string_value) = "request_headers"];
-  API_ATTRIBUTE_RESPONSE_HEADERS = 9 [(string_value) = "response_headers"];
-  API_ATTRIBUTE_REQUEST_BODY_SCHEMA = 10 [(string_value) = "request_body_schema"];
-  API_ATTRIBUTE_RESPONSE_BODY_SCHEMA = 11 [(string_value) = "response_body_schema"];
-  API_ATTRIBUTE_DEFINITION_TYPE = 12 [(string_value) = "api_definition_type"]; //deprecated. Use API_ATTRIBUTE_DISCOVERY_TYPE
-  API_ATTRIBUTE_DEFINITION = 13 [(string_value) = "api_definition"];
-  API_ATTRIBUTE_HTTP_PATH_PARAMS_TYPE = 14 [(string_value) = "http_path_params_type"];
-  API_ATTRIBUTE_HTTP_QUERY_PARAMS_TYPE = 15 [(string_value) = "http_query_params_type"];
-  API_ATTRIBUTE_HTTP_REQUEST_HEADERS_TYPE = 16 [(string_value) = "http_request_headers_type"];
-  API_ATTRIBUTE_HTTP_RESPONSE_HEADERS_TYPE = 17 [(string_value) = "http_response_headers_type"];
   API_ATTRIBUTE_IS_EXTERNAL_API = 18 [(string_value) = "IS_EXTERNAL_API"];
-  API_ATTRIBUTE_DEFINITION_FROM = 19 [(string_value) = "API_DEFINITION_FROM"]; //deprecated. Use API_ATTRIBUTE_DISCOVERY_FROM
-  API_ATTRIBUTE_HTTP_PATH_PARAMS_PII = 20 [(string_value) = "http_path_params_pii"];
-  API_ATTRIBUTE_HTTP_QUERY_PARAMS_PII = 21 [(string_value) = "http_query_params_pii"];
   API_ATTRIBUTE_IS_PII = 22 [(string_value) = "api_is_pii"];
   API_ATTRIBUTE_DISCOVERY_TYPE = 23 [(string_value) = "api_discovery_type"];
   API_ATTRIBUTE_DISCOVERY_FROM = 24 [(string_value) = "api_discovery_from"];
   API_ATTRIBUTE_DISCOVERY_STATE = 25 [(string_value) = "api_discovery_state"];
   API_ATTRIBUTE_DISCOVERY_TIMESTAMP = 26 [(string_value) = "api_discovery_timestamp"];
+  API_ATTRIBUTE_IS_STALE = 27 [(string_value) = "api_is_stale"];
```

Comment:
That's why I didn't add a new DISCOVERY state for INACTIVE, and created a new attribute

So, an API can be UNDER_DISCOVERY and INACTIVE, or DISCOVERED and INACTIVE
Also, an API can move from UNDER_DISCOVERY to DISCOVERED while being INACTIVE

Didn't want to add a new DISCOVERY state and treating it as a whole new API boolean attribute

Reply:
> Does this attribute represent an API entity state?

I cannot think of a use case where we would need a state transition for this. The attribute is a boolean attribute to mark stale API entities based on certain conditions

Reply:
How do we plan to consume the new attribute going ahead? Most likely it would be used in deciding the functional behaviour of API entity. As an example: if API is discovered but stale, then don't display the API entity on UI; don't calculate risk score; don't generate insights etc. In the current scope, we will end up revisiting all code paths with `if (state == DISCOVERED && !isStale)`. An alternative way is to promote it as a first-class entity state.

Reply:
How do we plan to consume the new attribute going ahead? Most likely it would be used in deciding the functional behaviour of API entity. As an example: if API is discovered but stale, then don't display the API entity on UI; don't evaluate any inferencing logic etc. In the current scope, we will end up revisiting all code paths with if (state == DISCOVERED && !isStale). An alternative way is to promote it as a first-class entity state.

Reply:
Makes sense. Coming to think of it, we don't need an extra attribute, and can just introduce a new API discovery state